src/integer_mod_q/mat_polynomial_ring_zq/arithmetic/mul_scalar.rs

```diff
@@ -0,0 +1,374 @@
+
+        assert_eq!(cmp_poly_ring_mat1, &poly_ring_mat1 * &integer1);
+        assert_eq!(cmp_poly_ring_mat2, &poly_ring_mat2 * &integer2);
+    }
+}
+
+#[cfg(test)]
+mod test_mul_poly_over_z {
+    use crate::integer::MatPolyOverZ;
+    use crate::integer::PolyOverZ;
+    use crate::integer_mod_q::MatPolynomialRingZq;
+    use crate::integer_mod_q::ModulusPolynomialRingZq;
+    use crate::integer_mod_q::PolynomialRingZq;
+    use std::str::FromStr;
+
+    /// Checks whether scalar multiplication is available for other types.
+    #[test]
+    fn availability() {
+        let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+        let poly_mat = MatPolyOverZ::from_str("[[3  0 1 1, 1  3],[0, 2  1 2]]").unwrap();
+        let poly_ring_mat = MatPolynomialRingZq::from((&poly_mat, &modulus));
+        let poly = PolyOverZ::from_str("2  1 1").unwrap();
+        let poly_ring = PolynomialRingZq::from((&poly, &modulus));
+
+        let _ = &poly_ring_mat * &poly_ring;
+        let _ = &poly_ring_mat * poly_ring.clone();
+        let _ = poly_ring_mat.clone() * &poly_ring;
+        let _ = poly_ring_mat.clone() * poly_ring.clone();
+
+        let _ = &poly_ring * &poly_ring_mat;
+        let _ = &poly_ring * poly_ring_mat.clone();
+        let _ = poly_ring.clone() * &poly_ring_mat;
+        let _ = poly_ring * poly_ring_mat;
+    }
+
+    /// Checks if scalar multiplication works.
+    #[test]
+    fn mul_correctness() {
+        let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+        let poly_mat1 = MatPolyOverZ::from_str("[[2  1 1, 1  3],[0, 2  1 2]]").unwrap();
+        let poly_ring_mat1 = MatPolynomialRingZq::from((&poly_mat1, &modulus));
+        let poly = PolyOverZ::from_str("2  1 1").unwrap();
+        let poly_ring = PolynomialRingZq::from((&poly, &modulus));
+
+        let poly_ring_mat2 = &poly_ring_mat1 * &poly_ring;
+
+        let cmp_poly_mat = MatPolyOverZ::from_str("[[3  1 2 1, 2  3 3],[0, 3  1 3 2]]").unwrap();
+        let cmp_poly_ring_mat = MatPolynomialRingZq::from((&cmp_poly_mat, &modulus));
+
+        assert_eq!(cmp_poly_ring_mat, poly_ring_mat2);
+    }
+
+    /// Checks if scalar multiplication reduction works.
+    #[test]
+    fn reduction_correct() {
+        let modulus = ModulusPolynomialRingZq::from_str("4  2 0 0 2 mod 17").unwrap();
+        let poly_mat1 = MatPolyOverZ::from_str("[[0, 1  10],[0, 2  1 2]]").unwrap();
+        let poly_ring_mat1 = MatPolynomialRingZq::from((&poly_mat1, &modulus));
+        let poly = PolyOverZ::from(2);
+        let poly_ring = PolynomialRingZq::from((&poly, &modulus));
+
+        let poly_ring_mat2 = &poly_ring_mat1 * &poly_ring;
+
+        let cmp_poly_mat = MatPolyOverZ::from_str("[[0, 1  3],[0, 2  2 4]]").unwrap();
+        let cmp_poly_ring_mat = MatPolynomialRingZq::from((&cmp_poly_mat, &modulus));
+
+        assert_eq!(cmp_poly_ring_mat, poly_ring_mat2);
```

Comment:
Please also include the polynomial reduction in this case. For that you need to use a multiplication with a polynomial that has degree at least 1 s.t. the degree of the initial PolyRingZq get's larger by multiplication. But in this case, you would need a polynomial like `3  1 0 2` to make the modulus as a polynomial being applied in the matrix.